lib/orbit/transformable.js

```diff
@@ -68,50 +71,34 @@ var Transformable = {
     if (object._transformable === undefined) {
       object._transformable = true;
       object._transformationQueue = new ActionQueue();
+      object._activeTransformations = [];
 
       Evented.extend(object);
 
-      object.didTransform = function(operation, inverse) {
-        var normalized = normalize(operation);
-        var transformation = transformationFor.call(this, normalized);
-        if (transformation) {
-          // console.log('Transformable#didTransform - matching transformation found', this.id, normalized, inverse);
-          transformation.pushCompletedOperation(normalized, inverse);
-
-        } else {
-          // console.log('Transformable#didTransform - createTransformation', this.id, normalized, inverse);
-          transformation = new Transformation(this);
-          transformation.pushCompletedOperation(normalized, inverse);
-          queueTransformation.call(this, transformation);
-        }
+      object.didTransform = function() {
+        Orbit.deprecate('`didTransform` has been deprecated. Please call `transformed` instead and supply a TransformResult as an argument.', true);
       };
 
-      object.currentTransformation = function() {
-        if (this._transformationQueue.current) return this._transformationQueue.current.data;
+      object.transformed = function(result) {
+        return this.applyTransformation( new Transformation(null, result) );
```

Comment:
Something else to consider, the flow in orbit-firebase for example is

    source.transform() => firebaseTransformer.transform()

and then 

    firebaseListener.on('changeInFirebase') => cache.transform() => source.emit('didTransform')

So the source is in effect split into two parts, the first receives transformations from UI and sends them to firebase, the second receives any changes in firebase (including those made by other clients), applies them to the cache and the source then emits them using didTransform.

Given this, this.applyTransformation can't replace didTransform because it's too far upstream in the process i.e. applyTransformation would be used to receive changes from the UI, not changes from firebase.